onnxruntime/core/providers/rocm/tunable/gemm_hipblaslt.h

```diff
@@ -0,0 +1,172 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+
+#ifdef USE_HIPBLASLT
+#include <hipblaslt/hipblaslt.h>
+#endif
+
+#include "core/common/common.h"
+#include "core/providers/rocm/tunable/gemm_common.h"
+#include "core/providers/rocm/tunable/rocm_tunable.h"
+
+namespace onnxruntime {
+namespace rocm {
+namespace tunable {
+namespace blas {
+namespace internal {
+
+#ifdef USE_HIPBLASLT
+
+template <typename T>
+constexpr hipblasDatatype_t HipBlasDataTypeFor(const T*) {
+  static_assert(sizeof(T) == 0, "Unsupported data type for hipBLASLt operation.");
+  // Compiler will complain if we don't return something.
+  return HIPBLAS_R_32F;
+}
+
+template <>
+constexpr hipblasDatatype_t HipBlasDataTypeFor(const float*) {
+  return HIPBLAS_R_32F;
+}
+
+template <>
+constexpr hipblasDatatype_t HipBlasDataTypeFor(const half*) {
+  return HIPBLAS_R_16F;
+}
+
+template <>
+constexpr hipblasDatatype_t HipBlasDataTypeFor(const BFloat16*) {
+  return HIPBLAS_R_16B;
+}
+
+template <>
+constexpr hipblasDatatype_t HipBlasDataTypeFor(const double*) {
+  return HIPBLAS_R_64F;
+}
+
+template <typename T, typename ParamsT>
+class HipBlasLtGemmOp {
+ public:
+  Status operator()(const ParamsT* params) {
+
+    hipblasLtHandle_t handle;
+    HIPBLASLT_CALL_THROW(hipblasLtCreate(&handle));
+
+    // Note: properties of original matrices A and B are swapped.
+    int64_t lda = (params->opb == BlasOp::N) ? params->n : params->k;
+    int64_t ldb = (params->opa == BlasOp::N) ? params->k : params->m;
+    int64_t ldc = params->n;
+    int64_t stride_a = params->n * params->k;
+    int64_t stride_b = params->k * params->m;
+    int64_t stride_c = params->n * params->m;
```

Comment:
This is not correct, `ld?` change be different from shape. `stride_?` should be `something * ld?`

Reply:
Could you elaborate? The original expression is like

```cpp
int64_t stride_a = (params->opb == BlasOp::N) ? lda * params->k : lda * params->n;
int64_t stride_b = (params->opa == BlasOp::N) ? ldb * params->m : ldb * params->k;
int64_t stride_c = ldc * params->m;
```

Reply:
Yeah, for None transposed `A`, `stride_a` should be computed as `lda * k`. It is critical to note you should use `lda` instead of `m` to do the math here, in case A's stride is non-trivial.